### PR TITLE
Caching: Store files to untrack and apply later

### PR DESCRIPTION
**Pull request by tulphoon**
### Checklist

- [ ] I have rebased properly. Please see [our tutorial on rebasing](http://coala.readthedocs.io/en/latest/Developers/Git_Basics.html#rebasing).
- [x] I have gone through the [commit guidelines](http://coala.readthedocs.io/en/latest/Developers/Writing_Good_Commits.html) and I've followed them.
- [x] I have followed the [guidelines for the commit shortlog](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#shortlog).
- [x] I have followed the [guidelines for the commit body](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#commit-body).
- [x] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#issue-reference).
- [ ] I have [run all the tests](http://coala.readthedocs.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!

### CODE REVIEW

**Comment by AbdealiJK** [comment]
@tulphoon as you see your tests are failing. Please check and fix. Also check the checklist in the PR message!

File: tests/misc/CachingTest.py [hunk 1]
```diff
@@ -130,3 +132,26 @@ def test_caching_results(self):
...
+        with bear_test_module():
+            with simulate_console_inputs("0"):
+                retval, output = execute_coala(
+                   coala.main,
+                   "coala",
+                   "-c", filename + ".coafile",
+                   "-f", filename + "test.py")
+                self.assertIn("This file has", output)
+
+            retval, output = execute_coala(
```
Comment (sils) [inline tests/misc/CachingTest.py]: why doesn't this need simulate console inputs? It should require the same inputs right?

**Comment by sils** [comment]
ack https://github.com/coala/coala/pull/3006/files

**Comment by sils** [comment]
@rultor merge

**Comment by sils** [comment]
ack 72a0112

**Comment by sils** [comment]
@rultor merge

**Pull request merged**